### ISSUE TITLE
Finetuning goes NAN after 1/2 epochs 

### ISSUE THREAD

**Issue by rikabi89** [SWivid/F5-TTS#341]
![image](https://github.com/user-attachments/assets/0c0181a8-0452-4154-a1ac-59fd679d80e1)

![{20FB7495-12C8-4AAD-9CBF-ABDA78F3A334}](https://github.com/user-attachments/assets/d4e14308-379f-4236-9e23-1c7db52794ca)

Any idea why this is happening? In this case it 1.5 hour dataset.

**Comment by rikabi89** [comment]
appears to be related to FP16? I switched it to "none" and everything was fine.

**Comment by SWivid** [comment]
@rikabi89 Sure, stuffs relative to gradient explosion
use bf16 or none in this case

**Comment by rikabi89** [comment]
> @rikabi89 Sure, stuffs relative to gradient explosion use bf16 or none in this case

Yes. But BF16 has another issue to do with a type in the name I think. I dont have the error right now as I'm training but it also doesnt work at all.

**Comment by rikabi89** [comment]
![{DB2BDE77-5BED-41E3-B5E5-61FA2E345AED}](https://github.com/user-attachments/assets/59aee4e7-5ec5-4b55-895f-ca9ed7d20408)
should be on f i guess

**Comment by lpscr** [comment]
@rikabi89 ops i update this i next update i miss this

**Comment by SWivid** [comment]
will close this issue, feel free to reopen if still not work with fp32 or bf16
